frontend/packages/operator-lifecycle-manager/src/components/modals/update-strategy-modal.tsx

```diff
@@ -0,0 +1,101 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { K8sKind, k8sPatch, K8sResourceKind, Patch } from '@console/internal/module/k8s';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalTitle,
+} from '@console/internal/components/factory/modal';
+import { withHandlePromise } from '@console/internal/components/utils';
+import {
+  ConfigureUpdateStrategy,
+  getNumberOrPercent,
+} from '@console/internal/components/modals/configure-update-strategy-modal';
+
+export const UpdateStrategyModal = withHandlePromise((props: UpdateStrategyModalProps) => {
+  const {
+    path,
+    resource,
+    resourceKind,
+    title,
+    handlePromise,
+    errorMessage,
+    inProgress,
+    cancel,
+    close,
+  } = props;
+  const getPath = path.replace(/\//g, '.');
+  const [strategyType, setStrategyType] = React.useState(
+    _.get(resource, getPath) || props.defaultValue,
+  );
+  const [maxUnavailable, setMaxUnavailable] = React.useState(
+    _.get(resource.spec, 'updateStrategy.rollingUpdate.maxUnavailable', '25%'),
+  );
+  const [maxSurge, setMaxSurge] = React.useState(
+    _.get(resource.spec, 'updateStrategy.rollingUpdate.maxSurge', '25%'),
+  );
+
+  const submit = (event) => {
+    event.preventDefault();
+
+    const patch: Patch = { path: '/spec/updateStrategy/rollingUpdate', op: 'remove' };
+    if (strategyType === 'RollingUpdate') {
+      patch.value = {
+        maxUnavailable: getNumberOrPercent(maxUnavailable || '25%'),
+        maxSurge: getNumberOrPercent(maxSurge || '25%'),
+      };
+      patch.op = 'add';
+    }
+
+    handlePromise(
+      k8sPatch(resourceKind, resource, [
+        patch,
+        { path: '/spec/updateStrategy/type', value: strategyType, op: 'replace' },
+      ]),
+    )
+      .then(close, () => {})
+      .catch((error) => {
+        throw error;
```

Comment:
The catch shouldn't be needed if we're only rethrowing the error

Reply:
> The catch shouldn't be needed if we're only rethrowing the error

Got eslint error when I removed catch - "error  Expected catch() or return  promise/catch-or-return"
I'm thinking of adding the "promise/catch-or-return": "off"  to eslint config file in operator-lifecycle-manager to omit the error or add a return statement to return the promise. Please let me know your thoughts. Thanks.

Reply:
> > The catch shouldn't be needed if we're only rethrowing the error
> 
> Got eslint error when I removed catch - "error Expected catch() or return promise/catch-or-return"
> I'm thinking of adding the "promise/catch-or-return": "off" to eslint config file in operator-lifecycle-manager to omit the error or add a return statement to return the promise. Please let me know your thoughts. Thanks.

Adding the "promise/catch-or-return": "off" to eslint config file in operator-lifecycle-manager to omit the error.

Reply:
> The catch shouldn't be needed if we're only rethrowing the error

Adding the "promise/catch-or-return": "off" to eslint config file in operator-lifecycle-manager to omit the error.